src/repo.rs

```diff
@@ -49,6 +49,43 @@ impl Repo {
             .collect()
     }
 
+    /// Transforms a git2::Branch into a git2::Comit
+    fn branch_to_commit(branch: git2::Branch) -> git2::Commit {
+        branch.into_reference().peel_to_commit().unwrap()
+    }
+
+    /// Walks trough revisions : returns all the ancestores Oids of a Commit
+    fn log(repo: &git2::Repository, commit: git2::Commit) -> Vec<git2::Oid> {
```

Comment:
I had no reason before, but I just tried and it changes performances a lot : 
I have 610 repositories and have ~20 seconds when I use `self.as_git2_repo()` and ~5 seconds when I use a reference. (on a 2 core 4 threads intel cpu).

I don't know how many repos people usually have on their computers (I have a lot of unused repos that I never delete), so this may not be an issue.

Maybe this could also be patched later with `as_git2_repo()` only running `git2::Repository::open` once  and storing the reference ?

Reply:
Wow, glad to hear you've been using this tool with hundreds of repos! I'm usually tracking only a dozen or so. I have no idea what's typical. Given that, no need to modify the call signature.

I imagine there are lots of ways this codebase could be optimized, including building the `::git2::Repository` in `new`. I probably didn't do it because not every command needs it, e.g., `list`, `scan`, and `info`. At this point, though, those are the exception. All the more useful commands end up calling `as_git2_repo()`, and I think that would be the norm for most subcommands added in the future. I'm not sure breaking the auto-derived traits is a deal-breaker, either. That said, it doesn't need to be addressed in this PR.